src/Symfony/Bridge/PhpUnit/ErrorAssert.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit;
+
+/**
+ * Test that your code triggers expected error messages.
+ *
+ * @author Christian Flothmann <christian.flothmann@xabbuh.de>
+ */
+final class ErrorAssert
+{
+    /**
+     * @param string[] $expectedMessages Expected deprecation messages
+     * @param callable $testCode         A callable that is expected to trigger the expected deprecation messages when being executed
+     */
+    public static function assertDeprecationsAreTriggered($expectedMessages, $testCode)
+    {
+        if (!is_callable($testCode)) {
```

Comment:
That's true but I'd like to keep it here too for two reasons: Do not confuse users by letting them deal with exceptions that have been thrown by methods they did not call if we cannot avoid that. Secondly, mitigate the risk of breaking anything if we refactor the method that we call here.